### PR TITLE
Create @markbind/core-web pkg

### PR DESCRIPTION
**Pull request by ang-zeyu**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Other, please explain: Extraction of frontend code and assets into a separate package

Helps slightly with #889 

**What is the rationale for this request?**
1. Further separate frontend concerns from the core package, hence the core package is now only concerned with the 'backend'
2. Having a single package for the frontend entry facilitates live/hot reload in a subsequent PR by consolidating the webpack build process

**What changes did you make? (Give an overview)**
1. Bootstrap the core web package (f2f4921) and Copy `asset/` over (fef3be3)
2. Shift the vue-components bundle build process over to core-web, adding the custom markbind setup scripts to it (`setup.js`)
   - babel - shifted babel config to the root, for use in both vue-components **tests** and core-web **build process** (dfd6e95)
   - package.jsons cleanup (18b52b2)
      - Made vue-components a `private` package, since the bundle is shifted to core-web, we don't need to publish it anymore
   - Setup entry points (imports, etc.) (dbbe9d8)
      - notably I removed `MarkBind.AfterSetup` here and moved it to `page.njk` (only for deprecation, we should remove this entirely sometime). This is unnecessary as layout scripts are already always loaded last (confirmed with some `console.log`s)
        - made this change here due to jQuery loading after `core.min.js` (and `core.min.js` using `jQuery` for the above mentioned `MarkBind.AfterSetup`) (a simple reorder would also fix it though, let me know if this should be removed here!)
   - Setup / move webpack configs from vue-components to core-web (4111113)
     - tell webpack to look for babel config in the root per dfd6e95
     - change in / output paths
     - Setup **css extraction** and minification, which is inline with how `markbind/page-nav/site-nav.css` is delivered. This also extracts css from the vue-components' components' `<style>...</style>`s, which should help slightly with fouc.
3. Adapt asset links in core package 169cfbe
4. Update docs 4907fd7

**Is there anything you'd like reviewers to focus on?**
- Should we still publish vue-components / core-web separately? This results in needing to write some extra `cp` scripts to copy over a separated bundle from core-web. (webpack does not support separate output paths for different entries by design). But I may have missed something out
- ❔ 

**Testing instructions:**
- test the netlify build
- test the build process `npm run build:web` from the root directory, and test update process `npm run updatetest(win)`

**Proposed commit message: (wrap lines at 72 characters)**
Create @markbind/core-web package

Various frontend assets and setup scripts used by the @markbind/core
package reside in the @markbind/core package itself.
This pollutes @markbind/core with frontend concerns.

To setup a live / hot reload process for frontend development, a single
consolidated webpack build process is also needed.

Hence, let's extract MarkBind's frontend scripts, styles and assets
into a separate package - @markbind/core-web.
This package would also house the consolidated webpack build process,
which also links in the @markbind/vue-components package.

### CODE REVIEW

**Comment by ang-zeyu** [comment]
rebased and updated tests with the added test page in #1324, also fixed some commit messages

File: packages/core/src/Page/page.njk [hunk 1]
```diff
@@ -57,4 +54,5 @@
 {%- endfor %}
 {%- endif %}
 <script src="{{ asset.layoutScript }}"></script>
+<script>if (MarkBind.AfterSetup) MarkBind.AfterSetup()</script>
```
Comment (marvinchin) [inline packages/core/src/Page/page.njk:57]: Can we leave a comment here to explain that this is for backward compatibility?

**Comment by marvinchin** [comment]
Please correct me if I'm wrong! Since we now generate a single bundle for our JS/CSS, does that mean that developers have to rebuild the bundle after making any changes to the source JS/CSS in `core-web`, and then re-running `markbind serve` before the changes will be reflected in the site? Is this something that will be addressed in #1327?

**Comment by ang-zeyu** [comment]
> Please correct me if I'm wrong! Since we now generate a single bundle for our JS/CSS, does that mean that developers have to rebuild the bundle after making any changes to the source JS/CSS in `core-web` (new additional step), and then re-running `markbind serve` before the changes will be reflected in the site? Is this something that will be addressed in #1327?

Thanks for looking at this!

Yes its addressed there, the difference is that even though currently we don't need to rebuild the frontend bundles for `markbind.css`, we still need to restart the server. 

Over in #1327 it would be edit -> changes reflected immediately

File: packages/cli/test/functional/test_site/expected/bugs/index.html [hunk 2]
```diff
@@ -14,10 +14,10 @@
   <link rel="stylesheet" href="../markbind/glyphicons/css/bootstrap-glyphicons.min.css">
   <link rel="stylesheet" href="../markbind/css/octicons.css">
   <link rel="stylesheet" href="../markbind/css/github.min.css">
-  <link rel="stylesheet" href="../markbind/css/markbind.css">
+  <link rel="stylesheet" href="../markbind/css/core.min.css">
```
[the remark this answers is not in the document]
Comment (ang-zeyu) [reply packages/cli/test/functional/test_site/expected/bugs/index.html]: yes that should be clearer!

File: packages/cli/test/functional/test_site/expected/bugs/index.html [hunk 3]
```diff
@@ -60,10 +60,9 @@
 </body>
 <script src="../markbind/js/bootstrap-utility.min.js"></script>
 <script>
-  const baseUrl = '/test_site'
-  const enableSearch = true
+  var baseUrl = '/test_site'
```
[the remark this answers is not in the document]
Comment (ang-zeyu) [reply packages/cli/test/functional/test_site/expected/bugs/index.html]: was testing something out, reverted 😅

File: packages/core-web/dist/js/core.min.js.LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/**!
```
[the remark this answers is not in the document]
Comment (ang-zeyu) [reply packages/core-web/dist/js/core.min.js.LICENSE.txt]: this was changed in #1300 and #1303, though its only observed now due to our update-bundle-at-release workflow

I extracted jquery out of vue-components in #1303, #1300 adds bootstrap vue into the build process so its dependency (popper.js) is observed here

**Comment by ang-zeyu** [comment]
if all is good I'll squash in the fixups and merge this! @marvinchin

**Pull request merged**